google/resource_dataproc_cluster_test.go

```diff
@@ -0,0 +1,780 @@
+	// Ensure it exists
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+		_, err := config.clientStorage.Objects.Get(bucket, object).Do()
+		if err != nil {
+			return fmt.Errorf("Unable to verify init action success: Error reading object %s in bucket %s: %v", object, bucket, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckDataproc_missingZoneGlobalRegion(rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "basic" {
+	name                  = "dproc-cluster-test-%s"
+	region                = "global"
+}
+`, rnd)
+}
+
+func testAccDataprocCluster_basic(rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "basic" {
+	name                  = "dproc-cluster-test-%s"
+	region                = "us-central1"
+
+}
+`, rnd)
+}
+
+func testAccDataprocCluster_basicWithAutogenDeleteTrue(rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "basic" {
+	name                  = "dproc-cluster-test-%s"
+	region                = "us-central1"
+
+	delete_autogen_bucket = true
+}
+`, rnd)
+}
+
+func testAccDataprocCluster_singleNodeCluster(rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "single_node_cluster" {
+	name   = "dproc-cluster-test-%s"
+	region = "us-central1"
+
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+
+    # Because of current restrictions with computed AND default
+    # [list|Set] properties, we need to add this empty config
+    # here otherwise if you plan straight away afterwards you
+    # will get a diff. If you have actual config values that is
+    # fine, but if you were hoping to use the defaults, this is
+    # required
+    master_config { }
+    worker_config { }
+}
+`, rnd)
+}
+
+func testAccDataprocCluster_withConfigOverrides(rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "with_config_overrides" {
+	name   = "dproc-cluster-test-%s"
+	region = "us-central1"
+
+	master_config {
+		num_masters       = 3
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+		num_local_ssds    = 0
+	}
+
+	worker_config {
+	    num_workers       = 3
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 11
+		num_local_ssds    = 0
+
+		preemptible_num_workers       = 1
+		preemptible_boot_disk_size_gb = 12
+	}
+}`, rnd)
+}
+
+func testAccDataprocCluster_withInitAction(rnd, bucket, objName string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "init_bucket" {
+    name          = "%s"
+    force_destroy = "true"
+}
+
+resource "google_storage_bucket_object" "init_script" {
+  name           = "dproc-cluster-test-%s-init-script.sh"
+  bucket         = "${google_storage_bucket.init_bucket.name}"
+  content        = <<EOL
+#!/bin/bash
+ROLE=$$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+if [[ "$${ROLE}" == 'Master' ]]; then
```

Comment:
do we actually use the difference between being on the master/worker anywhere in the test?